services/drupal/web/modules/custom/epa_workflow/src/Plugin/Block/EpaNodeDetailsBlock.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\epa_workflow\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\node\NodeInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+
+/**
+ * Provides an EPA Node Details block.
+ *
+ * @Block(
+ *   id = "epa_node_details",
+ *   admin_label = @Translation("EPA Node Details"),
+ *   category = @Translation("Custom"),
+ *   context_definitions = {
+ *     "node" = @ContextDefinition("entity:node", required = TRUE, label = @Translation("Node"))
+ *   }
+ * )
+ */
+class EpaNodeDetailsBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new EpaNodeDetailsBlock instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entityTypeManager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    // Get the node from the context.
+    $node = $this->getContextValue('node');
+
+    if ($node instanceof NodeInterface) {
+
+      // This loads the relationship entity that ties a node to a group.
+      $group_contents = \Drupal\group\Entity\GroupContent::loadByEntity($node);
```

Comment:
Given we have access to the entityTypeManager service above we can actually adjust this line to be this instead:
`$group_contents = $this->entityTypeManager->getStorage('group_content')->loadByEntity($node);` If you dive into that `loadByEntity()` method you'll see it's actually doing the exact same thing.